src/Symfony/Component/DependencyInjection/Tests/Compiler/AutowirePassTest.php

```diff
@@ -117,6 +117,23 @@ public function testTypeCollision()
         $pass->process($container);
     }
 
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage Unable to guess argument 1 for Symfony\Component\DependencyInjection\Tests\Compiler\NotGuessableArgument because the type-hinted class "Symfony\Component\DependencyInjection\Tests\Compiler\Foo" is used in more than one services.
+     */
+    public function testTypeNotGuessable()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('a1', __NAMESPACE__.'\Foo');
+        $container->register('a2', __NAMESPACE__.'\Foo');
+        $aDefinition = $container->register('a', __NAMESPACE__.'\NotGuessableArgument');
+        $aDefinition->setAutowired(true);
+
+        $pass = new AutowirePass();
+        $pass->process($container);
+    }
+
```

Comment:
You should add another test that is just like this, except that you explicitly call `set()` to map `\Foo` to service `a1`. You would assert that this would autowire `a1` and not throw an exception